From 77ef08c0174b9af97b12f60aa6bb29515d482efc Mon Sep 17 00:00:00 2001
From: Daniel Muey <dan@cpanel.net>
Date: Thu, 2 Jan 2020 08:08:42 -0600
Subject: [PATCH] Remove faulty systemd check from fpm sapi config

---
 sapi/fpm/config.m4 | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sapi/fpm/config.m4 b/sapi/fpm/config.m4
index ad89c1bb..fc06a05f 100644
--- a/sapi/fpm/config.m4
+++ b/sapi/fpm/config.m4
@@ -549,8 +549,6 @@ if test "$PHP_FPM" != "no"; then
     [no])
 
   if test "$PHP_FPM_SYSTEMD" != "no" ; then
-    PKG_CHECK_MODULES([SYSTEMD], [libsystemd >= 209])
-
     AC_CHECK_HEADERS(systemd/sd-daemon.h, [HAVE_SD_DAEMON_H="yes"], [HAVE_SD_DAEMON_H="no"])
     if test $HAVE_SD_DAEMON_H = "no"; then
       AC_MSG_ERROR([Your system does not support systemd.])
-- 
2.24.0

