From d6e644d2484afe7d14dbdbb83630c2cc5c91e853 Mon Sep 17 00:00:00 2001
From: Cory McIntire <cory@cpanel.net>
Date: Fri, 2 Mar 2018 11:10:07 -0600
Subject: [PATCH 06/11] FPM: Ensure docroot is in the user's homedir

---
 sapi/fpm/fpm/fpm_main.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/sapi/fpm/fpm/fpm_main.c b/sapi/fpm/fpm/fpm_main.c
index 342c9ab..dd3f3f7 100644
--- a/sapi/fpm/fpm/fpm_main.c
+++ b/sapi/fpm/fpm/fpm_main.c
@@ -636,6 +636,9 @@ static void php_cgi_ini_activate_user_config(char *path, int path_len, const cha
 
 	/* Check whether cache entry has expired and rescan if it is */
 	if (request_time > entry->expires) {
+		char pwnambuf[4096 + 256];
+		struct passwd pwd, *pwdp;
+		long homedirlen;
 		char * real_path;
 		int real_path_len;
 		char *s1, *s2;
@@ -664,6 +667,15 @@ static void php_cgi_ini_activate_user_config(char *path, int path_len, const cha
 			s_len = path_len;
 		}
 
+		/* Find our home directory, and ensure the document root is within it.
+		 */
+		if (getpwuid_r(geteuid(), &pwd, pwnambuf, sizeof(pwnambuf), &pwdp) || !pwdp || !pwdp->pw_dir)
+			return;
+
+		homedirlen = strlen(pwdp->pw_dir);
+		if (strncmp(doc_root, pwdp->pw_dir, homedirlen) || doc_root[homedirlen] != '/')
+			return;
+
 		/* we have to test if path is part of DOCUMENT_ROOT.
 		  if it is inside the docroot, we scan the tree up to the docroot
 			to find more user.ini, if not we only scan the current path.
-- 
2.24.1

